cfme/common/provider.py

```diff
@@ -728,3 +729,22 @@ def cleanup_vm(vm_name, provider):
     except:
         # The mgmt_sys classes raise Exception :\
         logger.warning('Failed to clean up VM %s on provider %s', vm_name, provider.key)
+
+
+def get_certificate(provider, sec_protocol=None, cert_source=None):
```

Comment:
please put it to BaseProvider class then. since it is only base class shared between containers and other providers.